doc/syntax.md

```diff
@@ -0,0 +1,476 @@
+
+When the sum above is used to describe conductances instead of currents, the function `cond_sum` can be used. It does exactly the same as `curr_sum` and can be used in exactly the same way and in the same cases, but makes explicit that the neural dynamics are based on synaptic conductances rather than currents.
+
+For modeling post synaptic responses with delta functions, `curr_sum` and `cond_sum` can be called with the keyword `delta` as first argument instead of a `shape` function.
+
+## Handling of time
+
+To retrieve some fundamental simulation parameters, two special functions are built into NESTML:
+
+* `resolution` returns the current resolution of the simulation in ms. This can be set by the user using the PyNEST function `nest.SetKernelStatus({"resolution": ...})`.
+* `steps` takes one parameter of type `ms` and returns the number of simulation steps in the current simulation resolution.
+
+These functions can be used to implement custom buffer lookup logic, but should be used with care.
+
+## Equations
+
+### Synaptic response
+
+A `shape` is a function of *t* (which represents the current time of the system), that corresponds to the shape of a postsynaptic response, i.e. the function *I*<sub>shape</sub>(*t*) with which incoming spike weights *w* are multiplied to compose the synaptic input *I*<sub>syn</sub>:
+
+<!--- $\large I_{\text{syn}}=\sum_{t_i\le t, i\in\mathbb{N}}\sum_{w\in\text{spikeweights}} w I_{\text{shape}}(t-t_i)$ --->
+![equation](https://latex.codecogs.com/svg.latex?%5Clarge%20I_%7B%5Ctext%7Bsyn%7D%7D%3D%5Csum_%7Bt_i%5Cle%20t%2C%20i%5Cin%5Cmathbb%7BN%7D%7D%5Csum_%7Bw%5Cin%5Ctext%7Bspikeweights%7D%7D%20w%20I_%7B%5Ctext%7Bshape%7D%7D%28t-t_i%29).
+
+### Systems of ODEs
+
+In the `equations` block one can define a system of differential equations with an arbitrary amount of equations that contain derivatives of arbitrary order. When using a derivative of a variable, say *V*, one must write: *V*'. It is then assumed that *V*' is the first time derivate of *V*. The second time derivative of *V* is *V*'', and so on. If an equation contains a derivative of order *n*, for example *V*<sup>(*n*)</sup>, all initial values of *V* up to order *n*-1 must be defined in the `state` block. For example, if stating
+```
+V' = a * V
+```
+in the `equations` block,
+```
+V mV = 0mV
+```
+has to be stated in the `state` block. If the initial values are not defined in `state` it is assumed that they are zero and unit checks are no longer possible.
+
+The content of spike and current buffers can be used by just using their plain names. NESTML takes care behind the scenes that the buffer location at the current simulation time step is used.
+
+## Dynamics and time evolution
+
+`update:` inside this block the current time can be accessed via the variable `t`
+
+> `integrate_odes`: Explain the current behavior
+
+> `process_spikes()`: Do we still have this function?
```

Comment:
Actually this one is something I proposed. The idea was to avoid having the user deal with spikes at the wrong time in the `update` block, but I don't know what its status is.
Moreover, since we talked about the possibility of making the ``update`` block optional by default, it might not be necessary in this case...